tests/expected/cpp/60057-issue_3116.cpp

```diff
@@ -0,0 +1,233 @@
+// Singular with various newline formats
+auto f = [] -> void {
+             return;
+         };
+
+auto f = [] -> void {
+             return;
+         }();
+
+auto f = [] -> void
+         {
+             return;
+         }();
+
+auto f =
+    [] -> void {
+        return;
+    };
+
+auto f =
+    [] -> void
+    {
+        return;
+    };
+
+auto f
+    = [] -> void {
+          int i = 0;
+          return;
+      };
+
+auto f
+    = []
+      {
+          int i = 0;
+          return;
+      };
+
+// Nested lambda
+auto f = [] {
+             auto g = [] {
+                          auto h = [] {
+                                       return;
+                                   };
+                          return;
+                      };
+             return;
+         };
+
+auto f = [] {
+             auto g = []
+                      {
+                          auto h = [] {
+                                       return;
+                                   };
+                          return;
+                      };
+             return;
+         };
+
+auto f = []
+         {
+             auto g = [] {
+                          auto h = []
+                                   {
+                                       return;
+                                   };
+                          return;
+                      };
+             return;
+         };
+
+// Nested lambda within functions
+Func(
+    [] {
+        return;
+    },
+    [] {
+        return;
+    }
+);
+
+Func([] {
+         return;
+     },
+     [] {
+         return;
+     }
+);
+
+Func([] {
+         return;
+     },
+     [] {
+         return;
+     }
+)();
+
+Func([] {
+         return;
+     },
+     [] {
+         return;
+     })();
+
+Func([] {
+         return;
+     },
+     [] {
+         return;
+     });
+
+A(
+    B([] (const std::string &s) -> bool {
+          s = "hello";
+          return true;
+      }), 1
+);
+
+A(
+    B(
+        [] (const std::string &s) -> bool {
+            s = "hello";
+            return true;
+        }
+    ), 1
+);
+
+// Inside scope
+{
+    std::thread([](const char *c) {
+                    std::cout << c << std::endl;
+                }).detach();
+
+    std::thread(
+        [](const char *c) {
+            std::cout << c << std::endl;
+        }
+    ).detach();
+
+    auto f = [&](int a) {
+                 return b;
+             };
+
+    auto f = [&](int a)
+             {
+                 return b;
+             };
+}
+
+Func(std::count_if(v.begin(), v.end(), [&](const auto &a) {
+                       return a == 3;
+                   }));
```

Comment:
Hmm... this seems a little sketchy, but it's consistent with 0.72, so I guess it's always been this way.